yew-macro/Makefile.toml

```diff
@@ -1,16 +1,25 @@
+[env]
+MSRV = "1.45.2"
+
+[tasks.install-msrv-toolchain]
+script_runner = "@duckscript"
+script = [
+    """
+    output = exec --fail-on-error rustup toolchain list
+    stdout = set ${output.stdout}
+    already_installed = contains ${stdout} ${MSRV}
+    if not ${already_installed}
+        echo "Installing toolchain for ${MSRV} (Yew MSRV) to run macro compilation tests"
+        exec --fail-on-error rustup toolchain install --profile minimal ${MSRV}
+    end
+    """
+]
 
 [tasks.test]
 clear = true
-dependencies = ["test-macro", "test-derive-props", "test-doc"]
-
-[tasks.test-macro]
-command = "cargo"
-args = ["test", "--test", "macro_test"]
-
-[tasks.test-derive-props]
-command = "cargo"
-args = ["test", "--test", "derive_props_test"]
-
-[tasks.test-doc]
-command = "cargo"
-args = ["test", "--doc"]
+dependencies = ["install-msrv-toolchain"]
+script = [
+    """
+    cargo +${MSRV} test
```

Comment:
@siku2 tasks can set the toolchain to work with using the toolchain attribute.
The value of that attribute is used by rustup to install if needed.
so you could do
```toml
[tasks.test]
toolchain = "1.45.2"
command = "cargo"
args = [ "test" ]
```

see more info at https://github.com/sagiegurari/cargo-make#usage-toolchain

Reply:
Oh wow, you even used it in the original version and I still totally missed it :sweat_smile:
I think I got confused by the `rust_version` condition which does something totally different.

I originally wanted to ask you for a review on this but I thought that would be an unreasonable thing to ask.
Anyway, thanks for pointing this out. That makes it a lot cleaner!

Reply:
@sagiegurari Finally got around to this! I noticed that unlike my little script, setting `toolchain` does not attempt to automatically download the toolchain.
The error it generates is nice enough so I'm still going to use it but cargo-make already automatically downloads packages (which takes significantly longer because they need to be compiled) so I would expect it to do the same for toolchains. What do you think?